honeybee/cli/create.py

```diff
@@ -0,0 +1,207 @@
+"""honeybee model creation commands."""
+
+try:
+    import click
+except ImportError:
+    raise ImportError(
+        'click is not installed. Try `pip install . [cli]` command.'
+    )
+
+from ladybug_geometry.geometry2d.pointvector import Point2D, Vector2D
+from ladybug_geometry.geometry2d.polygon import Polygon2D
+from ladybug_geometry.geometry3d.pointvector import Point3D, Vector3D
+from ladybug_geometry.geometry3d.face import Face3D
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+from ladybug_geometry_polyskel.polysplit import perimeter_core_subpolygons
+
+from honeybee.model import Model
+from honeybee.room import Room
+from honeybee.facetype import face_types as fts
+from honeybee.boundarycondition import boundary_conditions as bcs
+try:
+    ad_bc = bcs.adiabatic
+except AttributeError:  # honeybee_energy is not loaded and adiabatic does not exist
+    ad_bc = None
+
+import sys
+import os
+import logging
+import json
+import math
+import uuid
+
+_logger = logging.getLogger(__name__)
+
+
+@click.group(help='Commands for creating Honeybee models.')
+def create():
+    pass
+
+
+@create.command('shoe-box')
+@click.argument('width', type=float)
+@click.argument('depth', type=float)
+@click.argument('height', type=float)
+@click.option('--orientation-angle', help='A number between 0 and 360 for the clockwise'
+              'orientation of the box in degrees. (0 = North, 90 = East, 180 = South, '
+              '270 = West).', type=float, default=0, show_default=True)
+@click.option('--window-ratio', help='A number between 0 and 1 (but not equal to 1) '
+              'for the ratio between aperture area and area of the face pointing towards'
+              ' the orientation-angle.', type=float, default=0.4, show_default=True)
+@click.option('--adiabatic/--outdoors', help='Flag to note whether the faces that are '
+              'not in the direction of the orientation-angle are adiabatic or outdoors.',
+              default=True, show_default=True)
+@click.option('--units', help=' Text for the units system in which the model geometry '
+              'exists. Choose from (Meters, Millimeters, Feet, Inches, Centimeters).',
+              type=str, default='Meters', show_default=True)
+@click.option('--tolerance', help='The maximum difference between x, y, and z values '
+              'at which vertices are considered equivalent.',
+              type=float, default=0.01, show_default=True)
+@click.option('--output-file', help='Optional file to output the Model JSON string. '
+              'By default it will be printed out to stdout',
+              type=click.File('w'), default='-')
+def shoe_box(width, depth, height, orientation_angle, window_ratio, adiabatic,
```

Comment:
I can see that you have a separate command to add overhangs but I feel one of the most requested option will be adding horizontal/vertical shades by size and distance. Similar to the one that we support in Grasshopper.

Reply:
Yes, similarity to grasshopper is why it's in a separate command. The idea is that you can generate the base geometry with the command here and then pass it through some of the shade or window-generating commands, which have tons of customization options like Grasshopper.  I am breaking things down so that I don't get a single command that has 30+ inputs.